### PR TITLE
Fixed #24857 -- Added "python -m django" entry point.

### PR DESCRIPTION
**Pull request by ryanhiebert**
The entry point `django` would be a more obvious name than `django-admin`, and `python -m django` is a pretty obvious way to run the primary django command (`django-admin`). Add those methods of use as valid points of entry.

Note that I'm using "obvious" in my above description to mean that it's likely guessable by a newcomer to Django, not that it should have been so obvious that I'm complaining that it's not already added.

### CODE REVIEW

**Comment by timgraham** [comment]
I guess the question is whether or not it's worth having more than one way to do something. Perhaps you could raise the issue on the django-developers mailing list.

Related discussion: https://github.com/django/django/pull/3861

**Comment by apollo13** [comment]
-1 on another command -- the django-admin entrypoint was added to make live easier for windows users, but there is certainly no reason to add another script. I don't really care about ``__main__`` though.

**Comment by harrislapiroff** [comment]
Also -1 on another command. It feels very unpythonic to me to have two commands and seems like it will lead to more inconsistency and confusion in documentation. If I saw a `django` command mentioned in one part of the docs and `django-admin` command mentioned by another, I'd assume they were two different things (like `apache2` and `apache2ctl`). I have a hard time imagining a newcomer trying to run django without at least glancing at the documentation or knowing about tab completion...

**Comment by ryanhiebert** [comment]
OK, the consensus seems to be that adding a `django` command is not a good idea, but that adding the ability to run `python -m django` is fine. I've removed the former from the PR.

What tests does this need?

**Comment by timgraham** [comment]
In my testing, it seems the file should be added in `django/__main__.py`, not the repository root?

As for testing, maybe using the subprocess module would be the way to go? There's an existing example in `tests/model_regress/test_pickle.py`.

**Comment by ryanhiebert** [comment]
Yes indeed. Pure oversight, which highlights the needs for a test. Thanks for the reference, I'll write up a test.

**Comment by tricoder42** [comment]
Added ticket for this PR https://code.djangoproject.com/ticket/24857

**Comment by coderanger** [comment]
:+1: for the `-m django` format, would make my life much easier as a tool author that has to call django-admin in lots of weird automation stuffs.

File: django/__main__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+"""
+Alias to allow the django module to be run as a script.
+
+Example:
+
+    python -m django collectstatic
+"""
+from django.core import management
+
+if __name__ == "__main__":
```
Comment (apollo13) [inline django/__main__.py]: Isn't that a noop? This makes sense for normal console scripts, but here it will always be true, no?

Comment (ryanhiebert) [inline django/__main__.py]: Interestingly, no. If this module is imported from another package, it's name will be `django.__main__`. You could use that to, for example, allow you to have `django.__main__:management.execute_from_command_line` in the setup.py entry points without it executing it twice. (not that we'd want to do that here, but the pattern makes sense in other contexts)

**Comment by timgraham** [comment]
Ryan, are you able to add the test suggested in the ticket as well as documentation? If not, let's close the pull request and open it up for someone else to do so. Thanks!

**Comment by ryanhiebert** [comment]
That's fair. Time has been difficult to come by, but if you wouldn't mind giving me the weekend (Sunday, really) to see if I can make it happen. If it's not done then, go ahead a close it (if I don't get around to it).

**Comment by ryanhiebert** [comment]
@timgraham : Thanks for your prodding. I wasn't able to get to it yesterday, but thanks to Labor Day I had another chance, and I was able to take it.

**Comment by timgraham** [comment]
Would any documentation be useful? At least a mention in the release notes could be useful.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by ryanhiebert** [comment]
Yes, absolutely. Anywhere else that would be wise to have mention other than the release notes?

**Comment by timgraham** [comment]
merged in 617eff41acc583a3b0f97bb6bcc7efe096dc4891, thanks!

**Pull request closed without merging**